notary-server/src/server.rs

```diff
@@ -264,8 +260,104 @@ async fn load_tls_key_and_cert(
     Ok((private_key, certificates))
 }
 
+/// Load authorization whitelist if it is enabled
+fn load_authorization_whitelist(
+    config: &NotaryServerProperties,
+) -> Result<Option<HashMap<String, AuthorizationWhitelistRecord>>> {
+    let authorization_whitelist = if !config.authorization.enabled {
+        debug!("Skipping authorization as it is turned off.");
+        None
+    } else {
+        // Load the csv
+        let whitelist_csv = parse_csv_file::<AuthorizationWhitelistRecord>(
+            &config.authorization.whitelist_csv_path,
+        )
+        .map_err(|err| eyre!("Failed to parse authorization whitelist csv: {:?}", err))?;
+        // Convert the whitelist record into hashmap for faster lookup
+        let whitelist_hashmap = authorization_whitelist_vec_into_hashmap(whitelist_csv);
+        Some(whitelist_hashmap)
+    };
+    Ok(authorization_whitelist)
+}
+
+// Setup a watcher to detect any changes to authorization whitelist
+// When the list file is modified, the watcher thread will notify a tokio task to reload the whitelist
+// The watcher is setup in a separate thread by the notify library which is synchronous
+fn watch_and_reload_authorization_whitelist(
+    config: NotaryServerProperties,
+    authorization_whitelist: Option<Arc<Mutex<HashMap<String, AuthorizationWhitelistRecord>>>>,
+) -> Result<Option<RecommendedWatcher>> {
+    // Only setup the watcher if auth whitelist is loaded
+    let watcher = if let Some(authorization_whitelist) = authorization_whitelist {
+        // Use tokio unbounded channel sender to send event from synchronous watcher thread to async hot reload task
+        let (tx, mut rx) = mpsc::unbounded_channel();
+        // Setup watcher by giving it a function that will be triggered when an event is detected
+        let mut watcher = RecommendedWatcher::new(
+            move |event: Result<Event, Error>| {
+                match event {
+                    Ok(event) => {
+                        // Only send event if it's one that modified the file data
+                        if let EventKind::Modify(ModifyKind::Data(_)) = event.kind {
+                            debug!("Authorization whitelist is modified");
+                            tx.send(event).unwrap_or_else(|err| {
```

Comment:
Since we are using async mutex for whitelist, and notify is a sync lib — channel is used to bridge between sync and async worlds; it's also commonly used in notify repo examples (https://github.com/notify-rs/notify/tree/main/examples)